contracts/LiquidityPool.sol

```diff
@@ -276,13 +294,22 @@ contract LiquidityPool is Pausable, ReentrancyGuard, PiAdmin {
         _repay(msg.sender, msg.sender, _amount);
     }
 
+    function repayFor(address _account, uint _amount) external nonReentrant onlyDebtSettler {
```

Comment:
in this case the balance that will be used is the caller so, we can remove the `only` modifier
```suggestion
    function repayFor(address _account, uint _amount) external nonReentrant {
```